File: warmup/ruby/05-diagonal-difference.rb
#!/bin/ruby

n = gets.strip.to_i
a = Array.new(n)
for a_i in (0..n-1)
    a_t = gets.strip
    a[a_i] = a_t.split(' ').map(&:to_i)
end

def diagonal_diff(arr)
    diff = d1(arr) - d2(arr)
    diff.abs
end

def d1(arr)
    i = 0
    j = 0
    sum = 0

    while i < arr.length do
        sum = sum + arr[i][j]
        i = i + 1
        j = j + 1
    end

    sum
end

def d2(arr)
    i = 0
    j = arr.length - 1
    sum = 0

    while i < arr.length do
        sum = sum + arr[i][j]
        i = i + 1
        j = j - 1
    end

    sum
end

print diagonal_diff(a)

File: warmup/php/05-diagonal-difference.php
<?php
$handle = fopen ("php://stdin","r");
fscanf($handle,"%d",$n);
$a = array();
for($a_i = 0; $a_i < $n; $a_i++) {
    $a_temp = fgets($handle);
    $a[] = explode(" ",$a_temp);
    array_walk($a[$a_i],'intval');
}

function d1($arr) {
    $i = 0;
    $j = 0;
    $sum = 0;

    while ($i < count($arr)) {
        $sum += $arr[$i++][$j++];
    }

    return $sum;
}

function d2($arr) {
    $i = 0;
    $j = count($arr) - 1;
    $sum = 0;

    while($i < count($arr)) {
        $sum += $arr[$i++][$j--];
    }

    return $sum;
}

function diagonalDiff($arr) {
    return abs(d1($arr) - d2($arr));
}

echo diagonalDiff($a);
